src/contrasts.jl

```diff
@@ -553,12 +541,10 @@ mutable struct HypothesisCoding{T<:AbstractMatrix, S<:AbstractMatrix} <: Abstrac
     function HypothesisCoding(hypotheses::T;
                               levels::Union{AbstractVector,Nothing}=nothing,
                               labels::Union{AbstractVector,Nothing}=nothing) where {T<:AbstractMatrix}
-        labels == nothing && 
-            Base.depwarn(
-                "HypothesisCoding without specified contrast labels is deprecated.  " *
-                "Specify contrast labels with `HypothesisCoding(; labels=[...])` " *
-                "or HypothesisCoding(Dict(label1=>hyp1, label2=>hyp2, ...))",
-                :HypothesisCoding)
+        levels === nothing &&
+            throw(ArgumentError("kwarg `levels` must be specified for HypothesisCoding"))
+        labels === nothing &&
+            throw(ArgumentError("kwarg `labels` must be specified for HypothesisCoding"))
```

Comment:
Maybe just dropping the default value `nothing` would be enough, as the error message is quite explicit?